pkg/batcher/transport_message_batcher.go

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright 2025 Exactpro (Exactpro Systems Limited)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package batcher
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/rs/zerolog/log"
+	"github.com/th2-net/th2-common-go/pkg/queue/message"
+	transport "github.com/th2-net/transport-go/pkg"
+)
+
+type protocol = string
+
+const (
+	th2_pin_attribute string = "transport-group"
```

Comment:
issue(naming): according to golang guidelines we should use MixedCaps or mixedCaps instead of underscores
https://go.dev/doc/effective_go#mixed-caps

Also, since this is a constant, we should give it a more mindful name. This is not just a pin attribute, it is a transport protocol's pin attribute. So, maybe `th2TransportProtocolAttribute` or something similar will suit better here